mkosi/__init__.py

```diff
@@ -1245,7 +1293,7 @@ def configure_ssh(state: MkosiState) -> None:
     if not state.config.ssh:
         return
 
-    state.root.joinpath("etc/systemd/system/ssh.socket").write_text(
+    state.root.joinpath("usr/lib/systemd/system/ssh.socket").write_text(
```

Comment:
I think this change is in the right direction. I've been saying in the past, that we should put these into `/usr/lib/` instead of `/etc/`. That being said, we are directly clobbering stuff from the package manager here. There's a point in between: `/usr/local/lib`. The docs mention it for stuff the administrator installs (compared to what they create in `/etc/`), but in principle we could encroach there as the image builder and not step on the toes of package managers.

Reply:
The image comes from the same vendor as the packages, I don't think usr/local is appropriate

Reply:
I see your your point, but I think it depends a bit on the definition of vendor, whether one sees the distro or the image builder as vendor.  It's probably not much of an issue here, since these paths only come up in the context of `configure_ssh` and these images probably won't be shipped anywhere. We don't go out of our way for images to be upgradable, but generally I think if we can easily make that use case more seamless, by leaving all the package manager installed files alone (e.g. in Debian `/usr/lib/systemd/system/ssh.socket` is shipped by `openssh-server`), that's a good thing. 

I just generally prefer not to touch the distro stuff, if I can drop stuff in a higher level, but `/usr/local/`'s semantics are a bit unclear.

Reply:
Yeah /usr/local doesn't seem entirely appropriate for this stuff either.

I don't think we should make a big deal out of discussing this now. As soon as I can define dropins/units via credentials, I'm removing the --ssh option and this stuff completely, same for the autologin stuff.